src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java

```diff
@@ -64,10 +64,7 @@ public class RecordDeserializer {
   private static final ObjectMapper AVRO_OBJECT_MAPPER = new AvroMapper(new AvroFactory());
   public static final byte MAGIC_BYTE = 0x0;
   private static final Duration CACHE_FAILED_SCHEMA_ID_FETCH_DURATION = Duration.ofSeconds(30);
-  private static final Cache<Integer, String> SCHEMA_FETCH_ERROR_CACHE = Caffeine.newBuilder()
-      .expireAfterWrite(CACHE_FAILED_SCHEMA_ID_FETCH_DURATION)
-      .build();
-
+  private static SchemaErrors schemaErrors = new SchemaErrors();
```

Comment:
```suggestion
  @Inject
  private SchemaErrors schemaErrors;
```

We need to `@Inject` the `SchemaErrors` because it's a bean (see [this guide on dependency injection in Quarkus](https://quarkus.io/guides/cdi)). You can then remove the `schemaErrors` parameter from the constructor. Also, we don't need to make this variable `static` because there is only one instance of the `RecordDeserializer` (see this useful [blog post about bean scopes](https://marcelkliemannel.com/articles/2021/overview-of-bean-scopes-in-quarkus/#short-overview-of-quarkus-scopes)). We can look at it together when pairing later today.